File: sorting/SelectionSort.java
class SelectionSort{
    static int[] selectionSort(int[] A){
        for(int i=0;i<A.length;i++){
            int smallest = i;
            for(int j=i;j<A.length;j++){
                if(A[j]<A[i]){
                    smallest=j;
                }
            }
            int temp=A[i];
            A[i]=A[smallest];
            A[smallest]=temp;
        }
        return A;
    }
    public static void main(String[] args){
        int[] A={
            1, 10, 9, 6, 5, 3
        };
        A=selectionSort(A);
        for(int i=0;i<A.length;i++){
            System.out.print(A[i]+"\t");
        }
    }
}

File: sorting/SelectionSort.cpp
#include <iostream>

using namespace std;

int * selectionSort(int * A, int n, int x){
    for(int i=0;i<x;i++){
        int min=i;
        for(int j=i+1;j<n;j++){
            if(*(A+min)>*(A+j)){
                min=j;
            }
        }
        int temp=*(A+min);
        *(A+min)=*(A+i);
        *(A+i)=temp;
    }
    return A;
}

int main(){
    int n, x;
    cin>>n>>x;
    int Arr[n];
    for(int i=0;i<n;i++){
        cin>>Arr[i];
    }
    int * A=Arr;
    A=selectionSort(A, n, x);
    for(int i=0;i<n;i++){
        cout<<A[i]<<" ";
    }
    return 0;
}